src/main/scala/nl.knaw.dans.easy.bagstore/component/BagProcessingComponent.scala

```diff
@@ -175,11 +177,12 @@ trait BagProcessingComponent extends DebugEnhancedLogging {
       }
     }
 
-    def getReferenceBags(bagDir: Path): Try[Option[Path]] = Try {
+    def getReferenceBags(bagDir: Path, uuid: Option[UUID]): Try[Option[Path]] = Try {
       trace(bagDir)
       val refbags = bagDir.resolve("refbags.txt")
       if (Files.exists(refbags)) {
         // copy to tempDir
+        assertRefbagsFileIsNotEmpty(uuid, refbags).unsafeGetOrThrow
```

Comment:
instead of doing an `unsafeGetOrThrow` here, why don't you use a for-comprehension. If you look carefully, you see that we ignore the `Try` in `bagFacade.removeFromTagManifests` already, so that fixes that as well...

```scala
def getReferenceBags(bagDir: Path, uuid: Option[UUID]): Try[Option[Path]] = {
  trace(bagDir)
  val refbags = bagDir.resolve("refbags.txt")
  if (Files.exists(refbags))
    for {
      _ <- assertRefbagsFileIsNotEmpty(uuid, refbags)
      // copy to tempDir
      tempRefbags = Files.createTempFile(stagingBaseDir, "refbags-", "")
      _ = Files.deleteIfExists(tempRefbags)
      _ = Files.move(refbags, tempRefbags)
      _ = assert(!Files.exists(refbags), s"$refbags should have been moved to $tempRefbags, however, it appears to still be present here")
      // remove refbags.txt from all tagmanifests (if it was present there)
      _ <- bagFacade.removeFromTagManifests(bagDir, "refbags.txt")
    } yield Some(tempRefbags)
  else Success(None)
}
```